clean.py

```diff
@@ -72,24 +81,66 @@ def get_filtered_out_documents(
         logger.info(f"     Examples n°{idx} of filtered out examples:\n{filtered_out_ds[idx]}")
     return filtered_out_ds
 
-def apply_function(function_name: str, ds: Dataset, num_proc: int, batch_size: int, save_checks: bool) -> Tuple[Dataset, Optional[Dataset]]:
+def get_modified_documents(
+    ds: Dataset,
+    mapped_ds: Dataset,
+    num_proc: int,
+    batch_size: int,
+    sampling_size: Optional[int],
+    function_name: str
+) -> Dataset:
+    in_text_col_map, out_text_col_map = "old_text", "text"
+    remove_columns = set(ds.column_names)
+    remove_columns.remove("text")
+    ds = ds. remove_columns(remove_columns )
+    ds = ds.rename_column("text", "old_text")
+
+    if sampling_size is not None:
+        idx_samples = random.sample(range(len(ds)), min(len(ds), sampling_size))
+        ds = ds.select(idx_samples)
+        mapped_ds = mapped_ds.select(idx_samples)
+
+    mapped_diff_ds = concatenate_datasets([mapped_ds, ds], axis=1).filter(
+        partial(filter_diff_text, in_text_col=in_text_col_map, out_text_col=out_text_col_map),
+        batched=True, 
+        num_proc=num_proc, 
+        batch_size=batch_size
+    )
+    logger.info("Examples of modified examples:")
+    idx_samples = random.sample(range(len(mapped_diff_ds)), min(len(mapped_diff_ds), 10))
+    for idx in idx_samples:
+        logger.info(f"     Examples n°{idx} :\n{mapped_diff_ds[idx]}")
+
+    log_stats(f"Applied map function: {function_name}", len(ds), len(ds)-len(mapped_diff_ds), operation_type="Modified")
```

Comment:
Doesn't work, you've subsampled `ds` and `mapped_diff_ds` so this will be wrong if `sampling size is not None`